account_reconciliation_widget/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -1603,7 +1603,10 @@ odoo.define("account.ReconciliationModel", function (require) {
                     );
                     if (matching && matching.length == 2) {
                         matching = matching[1].replace(
-                            new RegExp("\\D" + reconcileModel.decimal_separator, "g"),
+                            new RegExp(
+                                "[^-d" + reconcileModel.decimal_separator + "]",
```

Comment:
In fact, this isn't correct anymore. Needs to be `"[^-\d" + reconcileModel.decimal_separator + "]",`